isyfact-standards-doc/src/docs/antora/modules/referenzarchitektur/pages/software-technisch/backend/spring.adoc

```diff
@@ -0,0 +1,290 @@
+Hierüber entstehen Abhängigkeiten zur Instanziierungsreihenfolge der Spring-Beans durch Spring.
+
+Für die Themen werden im folgenden Vorgaben aufgestellt.
+
+[[keine-beans-per-namen-auslesen]]
+=== Keine Spring-Beans per Namen auslesen
+
+Über den Anwendungskontext könnten Spring-Beans explizit per Namen ausgelesen werden.
+Dies ist mit einer Ausnahme verboten: Die Namen von Spring-Beans sollen nicht im Anwendungscode verwendet werden.
+Die Ausnahme gilt für den Zugriff von einem Anwendungskontext auf einen anderen (in Zusammenhang mit dem DispatcherServlet).
+In diesem Fall ist ein explizites Auslesen nicht zu vermeiden.
+Auszulesen ist in diesem Fall keine Komponente des Anwendungskerns, sondern eine weitere Schnittstellen-Bean, welche nur für diesen Zweck verwendet wird.
+
+[[aspektorientierte-programmierung-in-spring]]
+== Aspektorientierte Programmierung in Spring
+
+Es ist möglich, für Spring-Beans Funktionalität in Form von Aspekten zu definieren.
+Ihr Einsatz kann über _Pointcuts_ konfiguriert werden.
+Pointcuts definieren (etwa über reguläre Ausdrücke) Klassen und Methoden, welche um den Aspekt erweitert werden.
+
+Zu intensive Nutzung kann leicht zu einem schwer durchschaubaren Programmfluss führen.
+Deshalb soll AOP nur für folgende Bereichen genutzt werden: die Steuerung von Transaktionen, die Überwachung und die Berechtigungsprüfung.
+
+Explizit nicht benutzt werden soll AOP für die Fehlerbehandlung.
+
+Die Verwendung von AOP für andere Bereiche ist nur in begründeten Ausnahmefällen erlaubt.
+
+[[aop-fuer-transaktionssteuerung-verwenden]]
+=== AOP für Transaktionssteuerung verwenden
+
+Für die Transaktionssteuerung ist Spring-AOP mit den dafür vorgesehenen Klassen von Spring einzusetzen.
+Die Umsetzung geschieht im xref:software-technisch/backend/anwendungskern.adoc#transaktionssteuerung[Anwendungskern] oder in der xref:blaupausen:detailkonzept-komponente-service/master.adoc#transaktionssteuerung[Serviceschicht].
+Zusammengefasst gilt:
+
+* Instrumentiert werden alle Schnittstellenmethoden des Anwendungskerns.
+* Für jeden Aufruf des Anwendungskerns wird eine Transaktion gestartet.
+* Falls kein Fehler auftritt, wird die Transaktion abgeschlossen (Commit), sonst zurückgerollt (Rollback).
+
+[[aop-fuer-berechtigungspruefungen-verwenden]]
+=== AOP für Berechtigungsprüfungen verwenden
+
+Die Berechtigungsprüfung wird über Spring-AOP mit den vom xref:isy-security:nutzungsvorgaben/master.adoc[Baustein Security] angebotenen Annotationen umgesetzt.
+
+[[aop-nicht-fuer-das-logging-von-exceptions-verwenden]]
+=== AOP nicht für das Logging von Exceptions verwenden
+
+Sämtliche in einem Backend geworfenen und nicht behandelten Ausnahmen müssen inklusive ihrer Stack-Traces geloggt werden.
+Geloggt wird dies in den Methoden der Schnittstellen-Beans.
+Hierfür soll Spring-AOP nicht verwendet werden.
+
+Schnittstellen-Beans transformieren die Geschäftsobjekte des Anwendungskerns in Transportobjekte der Schnittstelle.
```

Comment:
Im Dokument anwendungskern.adoc ist (soweit ich mich erinnern kann) leider nirgends davon die Rede, dass es in der Schnittstelle (und in der Service-Schicht) Transport-Objekte geben soll